### ISSUE TITLE
Engineer wall timereduce determined by infected health

### ISSUE THREAD

**Issue by duralakun** [necropotame/teeworlds-infclass#180]
Right now the laser wall loses 1 sec for each zombie that dies in a wall. How about changing it so that the wall loses time depending on the health the zombie has. 
So If the zombie has full health (all hearts all armor) the wall loses 2 seconds. If the zombie has 10 health (full hearts no armor) the wall loses 1 second. If a zombie has 1 heart the wall loses 100 ms and so on.
If a zombie has 20 health and the wall has 1 second left, the zombie will not die but takes 10 damage and the wall disapears.
So basically for every hp a zombie has the wall loses 100ms. For every 100ms the wall gets reduced the zombie takes 1 hp damage.

**Comment by necropotame** [comment]
I don't like it because in the actual version, when I've only 1 heart, I prefer to suicide myself on a laser wall to be more useful to my team.

**Comment by duralakun** [comment]
I also like to do that. But i also want to jump in when i see the laser wall is becoming very thin.
Besides that you can hammer each other before jumping into the laser wall to be even more useful :)

**Comment by Socialdarwinist** [comment]
“Besides that you can hammer each other before jumping into the laser wall” – which would mean again that the engineer wall timereduce would be op, or up for low-health zombies if the higher timereduces are near the current timereduce. We have already decreased inf_barrier_timereduce from 2 seconds to 1.5 second, because 2 seconds was op.

**Comment by duralakun** [comment]
What was again the link for the statistics which class has the most kills? i lost it. I am not sure but i think engineers have by far the highest kills, which might suggest that they are op.

Anyways how about making it so that the min. time reduce is .. lets say 500ms, (5 hearts) so no matter if you have 1 or 3 hearts the time will be reduced by 500 ms so you can still be useful even with 1 heart

This system would also encourage humans behind the wall to do more damage to the zombies waiting.

**Comment by duralakun** [comment]
But dont take my numbers as exact numbers as how it should be, this could be balanced. Its more about the system that laser walls lose time depending on the zombies life.

**Comment by Socialdarwinist** [comment]
500 ms timereduce, and this for low-health zombies? I can’t believe that you just have suggested this (ૢ•᷄⊱•᷅ ╬). Grow some common sense instead of wheeling out devious “logical” rationalizations for full-blooded actuality!

The points repartition (currently not available) does not imply any power balance, for varied reasons:
* as it only shows registered people. Engineer-playing people may be more prone to register.
* it grandfathers scores from older times
* it reflect also other humans pulling in infected into walls
* there are conventional strategies that entail repartitions but are not ontologically necessary. One can achieve like scores defending with soldier, scientist, biologist and medic, or even sniper.
* …

You can much more easily break the game than balance its parts out with some intricate formula. People will be rightly grumpy if you add yet another imponderability in the engineer wall calculation. It is still desirable that skills decide more than – often mere coincidental – quantities.

But I guess you have problems with your logicism to get the fact the people’s skills are not merely a bundle of summed up contraptions, but have grown organically in opposition to cartesianisms. This game is not 2048.

**Comment by duralakun** [comment]
Socialdarwinist do you see any agression in what i write? I come here to discuss an idea peacefully, not to insult other people

**Comment by necropotame** [comment]
I agree, his anwser is completly exagerated. Socialdarwinist, please never do this again. It will just make other contributors leave.
